gon/shaped.py

```diff
@@ -435,13 +401,39 @@ def raw(self) -> RawPolygon:
         where ``vertices_count = len(self.border.vertices)\
  + sum(len(hole.vertices) for hole in self.holes)``.
 
-        >>> contour = Contour.from_raw([(0, 0), (1, 0), (0, 1)])
-        >>> contour.raw()
-        [(0, 0), (1, 0), (0, 1)]
+        >>> polygon = Polygon.from_raw(([(0, 0), (6, 0), (6, 6), (0, 6)],
+        ...                             [[(2, 2), (2, 4), (4, 4), (4, 2)]]))
+        >>> polygon.raw()
+        ([(0, 0), (6, 0), (6, 6), (0, 6)], [[(2, 2), (2, 4), (4, 4), (4, 2)]])
         """
         return self._raw_border[:], [raw_hole[:]
                                      for raw_hole in self._raw_holes]
 
+    def relate(self, other: Compound) -> Relation:
+        """
+        Finds relation between the polygon and the other geometric object.
+
+        Time complexity:
+            ``O(vertices_count * log vertices_count)``
+        Memory complexity:
+            ``O(vertices_count)``
+
+        where ``vertices_count = len(self.border.vertices)\
+ + sum(len(hole.vertices) for hole in self.holes)``.
+
+        >>> polygon = Polygon.from_raw(([(0, 0), (6, 0), (6, 6), (0, 6)],
+        ...                             [[(2, 2), (2, 4), (4, 4), (4, 2)]]))
+        >>> polygon.relate(polygon) is Relation.EQUAL
+        True
+        """
+        raw = self._raw_border, self._raw_holes
+        return (segment_in_polygon(other.raw(), raw)
+                if isinstance(other, Segment)
+                else (contour_in_polygon(other.raw(), raw)
+                      if isinstance(other, Contour)
+                      else polygon_in_polygon((other._raw_border,
+                                               other._raw_holes), raw)))
+
```

Comment:
And `Point`?

Reply:
`Point` is not a `Compound`, if we need to know whether is point inside polygon -- we should use `point in polygon` (i.e. `Polygon.__contains__` method)

Reply:
but it makes impossible to know whether the point is on the boundary or in interior of the polygon...

Reply:
the reason why I'm doing this only for `Compound`s is because they can have intersection with each other and `relate` method gives us analysis of its nature, point can't have intersection with other geometries because they consist of points, like number can't have intersection with numbers range, it doesn't make sense

Reply:
Could you elaborate on what exactly the `Compound` geometry is? Also, the fact that `Point` is a special case and requires a separate interface to check its relation to the target geometry object strikes me as a very peculiar and not an intuitive design.

Reply:
`Compound` consists of `Point`s, mathematically it's the set of points, so `Point` is a special case, because it's a building block, like empty set in set theory, it can't have elements because it is the basic element itself

Reply:
in the end `Compound` geometries will have an interface close to `Python`'s built in `set` (or to be precise -- `frozenset`) with ability to check if it has a point in it as an element, if it is disjoint with other `Compound` (set of points) and what is the result of set operations (intersection/union/difference/symmetric difference) for two `Compound`s

Reply:
it may be counter-intuitive at first, but makes sense from maths perspective I hope